rita_common/src/usage_tracker/mod.rs

```diff
@@ -112,18 +116,41 @@ pub struct UsageTracker {
 }
 
 impl UsageTracker {
-    fn save(&mut self) -> Result<(), IOError> {
-        let serialized = serde_json::to_vec(self)?;
+    fn save(&mut self) -> Result<(), RitaCommonError> {
+        let serialized = bincode::serialize(self)?;
         let mut file = File::create(settings::get_rita_common().network.usage_tracker_file)?;
-        let buffer: Vec<u8> = Vec::new();
-        let mut encoder = ZlibEncoder::new(buffer, Compression::default());
-        encoder.write_all(&serialized)?;
-        let compressed_bytes = encoder.finish()?;
-        file.write_all(&compressed_bytes)
+
+        let mut compressed_bytes = match compress_serialized(serialized) {
+            Ok(bytes) => bytes,
+            Err(e) => return Err(RitaCommonError::StdError(e)),
+        };
+
+        let mut newsize = MAX_TX_ENTRIES;
+        loop {
+            match file.write_all(&compressed_bytes) {
+                Ok(save) => {
+                    return Ok(save);
+                }
+                Err(e) => {
+                    // 500 tx min
+                    if newsize >= 1000 {
```

Comment:
we should add a comment here talking about how payments are larger than usage data.

Reply:
fixed